bot/cogs/spam/spam.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pathlib
+import time
+
+import discord
+from discord.ext import commands
+
+from ..variables import *
+
+spam_log_file = str(
+    pathlib.Path(
+        os.path.join(
+            os.path.abspath(os.getcwd()), "bot", "cogs", "spam", "spam_log.txt"
+        )
+    ).resolve()
+)
+
+
+class Spam(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+
+    @commands.Cog.listener()
+    async def on_message(self, message):
+        if message.author.id == self.client.user.id:
+            return
+
+        ctx = await self.client.get_context(message)
+
+        _timestamp = str(int(time.time()))
+        _authorid = str(message.author.id)
+        _channelid = str(message.channel.id)
+
+        _log = _timestamp + "___" + _authorid + "___" + _channelid
+
+        counter_messages = 0
+        counter_channels = 0
+        list_channels = []
+
+        with open(spam_log_file, "r+") as file:
+            file.writelines(f"{str(_log)}\n")
+
+            for line in file.readlines():
+                log_split = line.split("___")
+                log_timestamp = log_split[0]
+                log_authorid = log_split[1]
+                log_channelid = log_split[2]
+
+                # total number of messages sent by the user, in the last X seconds
+                if log_authorid == str(message.author.id):
+                    counter_messages += 1
+                    list_channels.append(log_channelid)  # add channel id
+
+                # number of unique channels the user has sent messages in
+                counter_channels = len(set(list_channels))
+
+        # print("{}, {}, {}".format(counter_messages, counter_channels, _log))
+
+        reason = "[POTENTIAL SPAM]: `{}` has sent `{}` messages in `{}` channels, within 5 seconds".format(
+            message.author.name, counter_messages, counter_channels
+        )
+
+        if counter_channels >= 3:
+            # if the use has sent messages to more than X unique channels - in Y seconds
+            # potential spam, ban user
+            await message.author.ban(reason=reason, delete_message_days=7)
```

Comment:
Based on the comment above about the X-seconds; 

Here I will get banned if I send 3 messages (potentially distinct) in 3 different channels. Which I guess is OK since we want to detect "faster than human" posting. Switching across 3 channels takes longer then 1 second, and we don't care if the messages are different then.

Reply:
Yep, super naive solution. Just logging to a file which is reset every 2 seconds
And it should be mostly safe to assume only a spam bot could send messages to 3+ channels within that time